docs/documentation/server_admin/topics/admin-console-permissions/fine-grain-v2.adoc

```diff
@@ -0,0 +1,374 @@
+
+| `view`                    | Defines if a realm administrator can view clients.
+| `manage`                  | Defines if a realm administrator can manage clients.
+| `map-roles`               | Defines if a realm administrator can assign any role defined by a client (or multiple clients) to a user. 
+| `map-roles-composite`     | Defines if a realm administrator can assign any role defined by a client (or multiple clients) as a composite to 
+                              another role.
+| `map-roles-client-scope`  | Defines if a realm administrator can assign any role defined by a client (or multiple clients) to the scope of 
+                              another client.
+|===
+
+[NOTE]
+====
+- An administrator must have explicit permission assigned for each operation to perform the corresponding action. For example, 
+  assigning only `manage` without `view` will prevent the client from being visible.
+- The `map-roles` operation does not grant the ability to manage users or assign roles arbitrarily; the administrator must also 
+  have user role mapping permissions.
+====
+
+===== Roles Resource Type
+
+TODO: review this section, I'm not sure how it should behave and if it's described correctly
+       - it seems view-realm role is needed to be able assign realm role
+
+The `Roles` realm resource type represents the roles in a realm. You can manage permissions for roles based on the following set of management operations:
+
+[cols="30%,70%"]
+|===
+| *Operation*               | *Description*
+
+| `map-role`               | Defines if a realm administrator can assign a role (or multiple roles) to a user.
+| `map-role-composite`     | Defines if a realm administrator can assign a role (or multiple roles) as a composite to another role. 
+                             Even if an administrator can manage client roles, they must have `map-role-composite` privileges for 
+                             the role they want to add as a composite.
+| `map-role-client-scope`  | Defines if a realm administrator can apply this role to a client’s scope. Even if they can manage 
+                             the client, they cannot create tokens for that client containing this role unless this privilege 
+                             is granted.
+|===
+
+[NOTE]
+====
+- The `map-role` operation does not grant the ability to manage users or assign roles arbitrarily; the administrator must also 
+  have user role mapping permissions. If an administrator has permission to `map-roles` for a client, permissions for specific 
+  client roles are not evaluated when mapping the role.
+- The `map-role-composite` operation ensures that administrators cannot add a role as a composite unless explicitly granted 
+  permission.  
+- The `map-role-client-scope` operation restricts the ability to apply roles to client scopes, even if the administrator has 
+  client management privileges.
+====
+
+==== Enabling admin permissions to a realm
```

Comment:
Replace ` with * everywhere you refer to something on the screen.   You only need ` for what you tell the user to type.